packages/components/molecules/f-searchbox/src/store/searchbox.module.js

```diff
@@ -234,13 +260,78 @@ export default {
                 : results.filter(addressDetails => addressDetails.type === 'Postcode');
         },
 
+        /**
+         *
+         * Updates the payload based on previously selected `Item.id` (From the retrieve API call).
+         *
+         * We also manually need to remove duplicate IDs from Loqate's response.
+         * (Bug in their response).
+         *
+         * @param state
+         * @param fullAddressDetails
+         */
+        [SET_FULL_ADDRESS_DETAILS]: (state, fullAddressDetails) => {
+            state.fullAddressDetails = fullAddressDetails.Items.reduce((seed, val) => {
+                if (!seed.some(s => s.Id === val.Id)) {
+                    seed.push(val);
+                }
+                return seed;
+            }, []).map(({
```

Comment:
All LGTM - but I've been reading over this for a little while and I don't think I fully have in my head what's happening.

I can see its for removing the duplicates but I think maybe because I don't understand what a `seed` is I'm struggling a little. I don't it needs any changes but we can have a quick chat 👍

Reply:
Sure, So seed here could be called anything (can change it if that helps?), here it's just represents each item in the `fullAddressDetails`. `val` represents each object in that array. 

Seed:
```javascript
[{Id: "GB|RM|ENG|7AA-RM10", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"}]
```

Val:

```javascript
{Id: "GB|RM|ENG|7AA-RM10", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"}
```

It will go through the full `fullAddressDetails` list and remove all of the duplicates based on the ID & feed that back to the map method. So if `seed` ended up having two objects with the same ID the reduce method would output:

Before reduce:
```javascript
[
 {Id: "GB|RM|ENG|7AA-RM10", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"},
 {Id: "GB|RM|ENG|7AA-RM10", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"},
 {Id: "GB|RM|ENG|7AA-AR51", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"}
]

After reduce:
`[
 {Id: "GB|RM|ENG|7AA-RM10", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"},
 {Id: "GB|RM|ENG|7AA-AR51", Type: "Postcode", Text: "RM10 7AA", Highlight: "0-4"}
]`

Let me know if that makes sense? :)

Reply:
Yes, that makes 💯  sense, really clear explanation thank you!